src/components/DataCollector/index.js

```diff
@@ -52,9 +50,10 @@ const createDataCollector = ({ config, logger, network }) => {
       });
   };
 
-  const createEventHandler = options => {
-    const event = createEvent();
-    const { viewStart = false, documentUnloading = false, xdm, data } = options;
+  const createEventHandler = (options, event = createEvent()) => {
+    const { viewStart = false, xdm, data } = options;
+    const documentUnloading =
+      options.documentUnloading || event.isDocumentUnloading();
```

Comment:
I like the addition of isDocumentUnloading() on the event, but for consistency we should probably set it on the event before calling onBeforeEvent, change the onBeforeEvent handlers to use that, and use it in makeServerCall.

```
if (options.documentUnloading) {
  event.documentUnloading();
}
return lifecycle
  .onBeforeEvent({ event, options, isViewStart: viewStart }) // notice I removed isDocumentUnloading
  .then(() => {
    ...
  })
  .then(() => makeServerCall(event)) // use the event.isDocumentUnloading() inside makeServerCall
};
```